File: dmy_house_logs/perl/log_tail.pl
#! /usr/bin/env perl

use strict;
use warnings;

use CGI::Fast;
use tmpl_log_tail;

my $LOG_TAIL_LEN = 5;

my @segs = split /\//, $0;
my $tmpl_name = $segs[-1];
$tmpl_name =~ s/_log_tail\.cgi$/.html/;

while (my $q = new CGI::Fast) {
    gen_tmpl_log_tail($LOG_TAIL_LEN, $tmpl_name, $q);
}

File: dmy_house_logs/fast/log_tail.c
#include <string.h>

#include "tmpl_log_tail.h"

#define LOG_TAIL_LEN 5

int
main (int argc, char *argv[])
{
  char tmpl_name[260];
  if (!gen_tmpl_name (tmpl_name, 260, argv[0])) {
    return 1;
  }
  gen_tmpl_log_tail (LOG_TAIL_LEN, tmpl_name);
  return 0;
}
